java-manta-client-unshaded/src/main/java/com/joyent/manta/http/EncryptionHttpHelper.java

```diff
@@ -534,7 +537,13 @@ public MantaObjectResponse httpPutMetadata(final String path,
          * the encryption-specific metadata headers. While at the same time
          * overwriting all other metadata values. Unfortunately, this process
          * requires two steps. */
-        HttpResponse response = httpHead(path);
+        String metadataPath = path;
+        if (StringUtils.endsWith(path, SEPARATOR + "metadata")) {
+                    metadataPath = StringUtils.substringBeforeLast(path, SEPARATOR);
+        }
+
+        Validate.notNull(metadataPath, "Http Response cannot be retrieved for an empty path");
```

Comment:
Are you checking to see if the metadata path is empty or null? Your error message implies empty.

If you want to check for not empty, make use of the `Validate.notEmpty(T object, String message, Object... values)` signature that will let you format your message so that you can know what type of an empty string you got passed - null vs. zero byte string.

Reply:
Noted.